var/spack/repos/builtin/packages/py-numpy/package.py

```diff
@@ -102,6 +103,7 @@ class PyNumpy(PythonPackage):
     depends_on('python@3.6:3.10', type=('build', 'link', 'run'), when='@1.19')
     depends_on('python@3.7:3.10', type=('build', 'link', 'run'), when='@1.20:1.21')
     depends_on('python@3.8:', type=('build', 'link', 'run'), when='@1.22:')
+    depends_on('py-packaging', when='@1.21: platform=darwin target=aarch64:', type='build')
```

Comment:
Where does this new requirement come from?

Reply:
From the pyproject.toml

Reply:
Ah, there's this:
```
"packaging==20.5; platform_machine=='arm64'",  # macos M1
```
That is only for that version, and while I forgot the reason for it, it should not be needed for newer releases.

Reply:
It looks like it's been there since 1.21 and it's still there in main

Reply:
The requirement is there, but it's `==20.5` which is different from `@1.21:`. The dependency is not needed for `>= 1.21`.

Reply:
Argh, I misread that. Okay now I'm curious where it came from, `numpy` builds just fine without `packaging`. So your change here matches what's in numpy main, but I believe it shouldn't be there.

Reply:
Looks like https://github.com/numpy/numpy/pull/18538

Reply:
Indeed no longer needed, see https://github.com/numpy/numpy/pull/21963

Reply:
Ah, thanks for clearing that up. The reason I didn't need this in Spack on M1 is because we never pinned setuptools. We still don't support separate concretization of build-only dependencies so we won't be able to pin for a while. Not sure if we should keep this in here for the sake of matching the pyproject.toml in case we pin these in the future or if we should just remove. Any opinions?

Reply:
I'd remove this, because it is just a dependency that isn't needed at all.